### PR TITLE
(feat) O3-3544: Add search bar to the orders table

### PR DESCRIPTION
**Pull request by arodidev**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR introduces search functionality to the orders table by adding a search bar.

## Screenshots

https://github.com/user-attachments/assets/b043e675-35b7-413f-9352-228ac4658154

## Related Issue

https://openmrs.atlassian.net/browse/O3-3544

### CODE REVIEW

**Comment by pirupius** [comment]
The UI doesn't seem quite right. Let's wait for [feedback from the design team](https://openmrs.slack.com/archives/CKS32D55G/p1721985286836339)

**Comment by pirupius** [comment]
@arodidev did u make the recommended UI changes? If so please update the description or ping me so i can take a look

**Comment by arodidev** [comment]
Hi @pirupius yes, kindly refer to the screenshot above and advise if this suffices. I believe we still haven't gotten a clear way forward from Paul and Ciarán [regarding the issue of including filtering on the table](https://openmrs.slack.com/archives/CKS32D55G/p1722332789664479?thread_ts=1721985286.836339&cid=CKS32D55G).

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 1]
```diff
@@ -21,6 +21,9 @@ import {
   TableHeader,
   TableRow,
   Tooltip,
+  Layer,
```
Comment (denniskigen) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:24]: nit: It'd be nice to keep this list alphabetically ordered for easier scanning.

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**